src/werkzeug/routing.py

```diff
@@ -1052,8 +1052,9 @@ def compile(self, append_unknown=True):
                 # resolve it to a constant ahead of time
                 if is_dynamic and data in self.defaults:
                     data = self.rule._converters[data].to_url(self.defaults[data])
-                    is_dynamic = False
-                if not is_dynamic:
+                    opl.append((None, data))
+                    continue
+                elif not is_dynamic:
```

Comment:
I think this is unrelated to the patch (?)

I changed this in my branches to be `if, elif, else` and dropped the `continue`s :man_shrugging:

Reply:
Seems to be required for the tests to pass. I'll just overwrite it when I get to your PR.

Reply:
oh yes, I see it now :+1: -- I can rebase mine too once this merges as well